Doc/library/typing.rst

```diff
@@ -572,6 +572,32 @@ These can be used as types in annotations and do not support ``[]``.
    * Every type is compatible with :data:`Any`.
    * :data:`Any` is compatible with every type.
 
+.. data:: Never
+
+   Notation for the `bottom type <https://en.wikipedia.org/wiki/Bottom_type>`_,
+   a type that has no members.
+
+   This can be used to define a function that should never be
+   called, or a function that never returns::
+
+     from typing import Never
+
+     def never_call_me(arg: Never) -> None:
+         pass
+
+     never_call_me(1)  # type checker error
+
+     def stop() -> Never:
+         return 1  # type checker error
```

Comment:
It feels wrong to only show examples of errors. Please give a positive example too (e.g. a function that raises immediately). Maybe only give a positive example?

Reply:
Changed this to also give the `int_or_str` example from the `assert_never` docs.